src/decode-pppoe.c

```diff
@@ -185,46 +203,45 @@ int DecodePPPOESession(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p,
 
             case PPP_VJ_UCOMP:
 
-                if(len < (PPPOE_SESSION_HEADER_LEN + IPV4_HEADER_LEN))    {
+                if (len - pppoesh_len < IPV4_HEADER_LEN) {
```

Comment:
Why did you move `pppoesh_len` to the other side of the comparison ?

Reply:
It's to prevent a warning: comparison of integer expressions of different signedness: ‘uint32_t’

Reply:
Ok.
I am not sure I understand the implicit integers conversions here, so I will trust the compiler